solr/core/src/java/org/apache/solr/cluster/placement/Replica.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.solr.cluster.placement;
+
+/**
+ * An instantiation (or one of the copies) of a given {@link Shard} of a given {@link SolrCollection}.
+ * Objects of this type are returned by the Solr framework to the plugin, they are not directly built by the plugin. When the
+ * plugin wants to add a replica it goes through appropriate method in {@link PlacementPlanFactory}).
+ */
+public interface Replica extends PropertyValueSource {
+  Shard getShard();
+
+  ReplicaType getType();
+  ReplicaState getState();
+
+  String getReplicaName();
+  String getCoreName();
+
+  /**
+   * {@link Node} on which this {@link Replica} is located.
+   */
+  Node getNode();
```

Comment:
Do we need a Node instance or just a node name here?

Reply:
We're going to manipulate `Node` instances in the plugin code. Therefore it's easier that all concepts of `Node` are the nodes themselves and not just their names. Current implementation only defines the name on the node, but this can change. Nodes are also used as targets for fetching values (`PropertyValueSource`), so it's convenient to have them implement the same interfaces as other targets. On the other hand, most key building ended up accepting a `Node` directly, and I don't know if key that are not implemented yet will require flexibility to be applicable to either `Node`, `Shard`, `SolrCollection` or `Replica` transparently.
I found it more consistent and easier to read to manipulate `Nodes` rather than `Strings` basically if you suggestion is to get rid of `Node` completely.